test/services/ValidateIncorporatedEntityDetailsServiceSpec.scala

```diff
@@ -53,14 +53,23 @@ class ValidateIncorporatedEntityDetailsServiceSpec extends AnyWordSpec with Matc
 
         mockGetCtReferenceConnector.getCtReference(eqTo(testCompanyNumber))(eqTo(hc)) returns Future.successful(Some(testCtReference))
 
-        await(TestValidateIncorporateEntityDetailsService.validateDetails(testCompanyNumber, mismatchedTestCtReference)) mustBe DetailsMismatched
+        await(TestValidateIncorporateEntityDetailsService.validateDetails(testCompanyNumber, Some(mismatchedTestCtReference))) mustBe DetailsMismatched
       }
     }
     s"return $DetailsNotFound" when {
       "there is no stored CT Reference for the provided Company Number" in {
         mockGetCtReferenceConnector.getCtReference(eqTo(testCompanyNumber))(eqTo(hc)) returns Future.successful(None)
 
-        await(TestValidateIncorporateEntityDetailsService.validateDetails(testCompanyNumber, testCtReference)) mustBe DetailsNotFound
+        await(TestValidateIncorporateEntityDetailsService.validateDetails(testCompanyNumber, Some(testCtReference))) mustBe DetailsNotFound
+      }
+    }
+
+    s"return $DetailsNotFound" when {
+      "the user asserts the unincorporated association does not have a Ct Utr, but one is found" in {
+
+        mockGetCtReferenceConnector.getCtReference(eqTo(testCompanyNumber))(eqTo(hc)) returns Future.successful(Some(testCtReference))
```

Comment:
Hey Richard I dont think eqTo is required because the equality is always calculated properly without it. I know that it was there already unfortunately .... if you want you can leave it .... if you want you can remove it because it is not necessary .... up to you

Reply:
Done